microservices/optimizer/pyproject.toml

```diff
@@ -9,6 +9,7 @@ package-mode = false
 [tool.poetry.dependencies]
 python = "3.12.2"
 Flask = "3.0.3"
+pymongo = "4.6.3"
```

Comment:
Switch this to a wildcard (I fixed the others in the Poetry PR up right now, my fault)
```suggestion
pymongo = "*"
```

Reply:
I don't agree with this. * can lead to sudden issues if dependency updates mess with other dependencies or env vars or our code. We should have to manually update if we see the need

Reply:
The actual version of the dependency that is installed is in `poetry.lock` which remains stagnant. dependabot's version updates will create PRs to update the dependencies as the packages are updated, which we can accept as we see fit. Despite it being a wildcard here, the dependencies will never update on their own since this will require changing the lock file, not this file. It's a quirk of the Poetry lock system

Reply:
that not how poetry supposed to work. Users are supposed to run `poetry install` whenever they download the repo and everytime we add a new dependency we should be running `poetry install`. The lock file being checked into the repo is for reference incase our dependencies aren't working. Everyone's local copy should be changed through `poetry install` as needed.

Reply:
devop rule #1: never automatically update packages 😎

Reply:
Discussed on discord. Dependabot is broken without this. Changed.